### DISCUSSION TITLE
Arbitrary milestones!

### DISCUSSION THREAD

**Discussion by wiredfool** [python-pillow/Pillow#6636] [General]
We're at 2000 stars! Seems like only last month that we were at 1000.

**Comment by hugovk** [comment]
10,511,869 downloads!
784k monthly downloads!
71,761 total lines!
45,727 lines of code!
22,623 lines of C!
22,221 lines of Python!
15,429 monthly views!
13,457 blank lines!
12,577 lines of comments!
6,822 monthly unique visitors!
4,053 commits!
2,613 Travis CI builds!
894 pull requests!
647 monthly clones! 
618 issues!
438 forks!
179 monthly unique clones!
145 contributors!
116 watchers!
31 releases!

**Comment by hugovk** [comment]
:100::100::100::100::100: forks!

![500 forks](https://cloud.githubusercontent.com/assets/1324225/14249768/06d63b40-fa84-11e5-9b9b-d8c16a78aae5.png)

13,023,183 downloads!

**Comment by aclark4life** [comment]
Hah, only 33K more to catch up to [Oh My ZSH](https://github.com/robbyrussell/oh-my-zsh)

**Comment by hugovk** [comment]
We're at 3,000 stars! Seems like only last month that we were at 2,000.

![image](https://cloud.githubusercontent.com/assets/1324225/21091065/982b2676-c04c-11e6-85c9-c436964f130c.png)

**Comment by radarhere** [comment]
50 releases

**Comment by hugovk** [comment]
We're at 4,000(+6) stars! Seems like only last month that we were at 3,000.

![image](https://user-images.githubusercontent.com/1324225/31277334-1b41b466-aaa9-11e7-9f8c-9665a8d0e6a4.png)

**Comment by hugovk** [comment]
7,000 commits!

![screenshot_20180128-094205](https://user-images.githubusercontent.com/1324225/35480090-cd26f03c-040f-11e8-9dbc-92055f5bb6f4.jpg)

**Comment by aclark4life** [comment]
Seems like we may want to start celebrating Pillow's birthday each year on [July 31, 2010](https://pypi.org/project/pillow/1.0/). :smile:

**Comment by hugovk** [comment]
Here's your announcement to [Image-SIG](https://mail.python.org/pipermail/image-sig/2010-July/006423.html) on July 31, 2010:

> **[Image-SIG] Pillow, the "friendly" PIL fork**

> I have been wanting to tackle this for sometime, and finally got around to it tonight. I've created a "friendly" fork of PIL called "Pillow": http://pypi.python.org/pypi/Pillow/1.0

And [from Fredrik](https://mail.python.org/pipermail/image-sig/2005-June/003377.html
), showing PIL's birthday as June 15, 1995:

> **[Image-SIG] pil ten year anniversary**

> just fyi:
>
> $ cd pil
$ grep 1995-06 */*.c
libImaging/Convert.c: * 1995-06-15 fl   created
libImaging/Except.c: * 1995-06-15 fl   Created
libImaging/Geometry.c: * 1995-06-15 fl  Created
libImaging/Histo.c: * 1995-06-15 fl   Created.
libImaging/Storage.c: * 1995-06-15 fl   Created
>
> thanks to everyone who's used PIL for cool stuff for the last decade,
and everyone who's contributed cool stuff to PIL !
>
> cheers /F

**Comment by radarhere** [comment]
3,000 issues/PRs

**Reply by hugovk** [reply]
8,000 issues/PRs!

https://github.com/python-pillow/Pillow/pull/8000

**Reply by radarhere** [reply]
10,000 issues/PRs

**Comment by radarhere** [comment]
1,001 forks

**Comment by radarhere** [comment]
200 watchers

**Comment by hugovk** [comment]
We're at 5,000(+40) stars!

![image](https://user-images.githubusercontent.com/1324225/42138084-7195edd2-7d80-11e8-85b2-2bf6152bc671.png)

**Comment by hugovk** [comment]
Pillow's pip installs from PyPI, showing Python 2 has recently dropped under the 50% mark: 

![pip-install-pillow](https://user-images.githubusercontent.com/1324225/42414730-32fad4a2-8244-11e8-9384-2cbb3dfc7822.png)

https://medium.com/@hugovk/python-version-share-over-time-cf4498822650

**Comment by aclark4life** [comment]
May as well leave this one open until we no longer intend to celebrate arbitrary milestones, which is never.

**Comment by aclark4life** [comment]
**88 Following & 91 Followers!** Lower Following count due to 3 private accounts that follow us but as of yet, have not responded to python-pillow's follow request.

![screen shot 2018-07-08 at 7 50 00 am](https://user-images.githubusercontent.com/72164/42419465-9f9510de-8283-11e8-88d7-5748fc7692f4.png)

**Comment by hugovk** [comment]
**5,555 stars!**

![image](https://user-images.githubusercontent.com/1324225/49027425-d2e7b380-f1a8-11e8-8026-f660bd16710c.png)

**Comment by hugovk** [comment]
**6,000+1 stars!**

![image](https://user-images.githubusercontent.com/1324225/55160617-67b8e880-516c-11e9-8897-cdd8f771dc9d.png)

**Comment by aclark4life** [comment]
Via [GitHub Projects](https://github.com/python-pillow/Pillow/projects/1?fullscreen=true) & my agile/drag/drop skills, **76 open issues not updated this year (Icebox)**, while **97 open issues have received at least a comment in 2019 (In Progress)**!

<img width="1680" alt="Screenshot 2019-05-11 14 02 49" src="https://user-images.githubusercontent.com/72164/57573439-82ca7980-73f5-11e9-8081-b9b485ae2053.png">

**Comment by radarhere** [comment]
Used by 136,898 repositories

![Screen Shot 2019-05-27 at 10 52 02 pm](https://user-images.githubusercontent.com/3112309/58420957-1082b600-80d2-11e9-8138-ea2db948b0d2.png)

**Comment by radarhere** [comment]
Over 250 contributors

**Comment by aclark4life** [comment]
4000th PR! (noted by @hugovk in https://github.com/python-pillow/Pillow/pull/4000)

**Comment by hugovk** [comment]
Technically 4,000th PR or issue, the same serial number is used for both:

* 1,639 issues
* 2,455 PRs

**Comment by radarhere** [comment]
10 million downloads per month

**Comment by hugovk** [comment]
![image](https://user-images.githubusercontent.com/1324225/74934444-e8b94800-53ee-11ea-88e1-92119dadb3ac.png)

> Downloads last day: 434,547
> Downloads last week: 2,406,242
> Downloads last month: 10,065,081

https://pypistats.org/packages/pillow

**Comment by aclark4life** [comment]
@radarhere @hugovk Crazy!!

**Comment by radarhere** [comment]
10,000 commits

**Comment by aclark4life** [comment]
56 people like the new Facebook Pillow page!

<img width="995" alt="Screen Shot 2020-05-21 at 6 06 36 AM" src="https://user-images.githubusercontent.com/72164/82548492-5b8aaa00-9b29-11ea-928d-e66f305717a4.png">

**Comment by aclark4life** [comment]
Also, first time we've broken 100 views (with 140) to our Tidelift package URL (https://tidelift.com/subscription/pkg/pypi-pillow)! Been watching it hover just under 100 forever … 🎉 

<img width="849" alt="Screen Shot 2020-05-22 at 7 15 11 AM" src="https://user-images.githubusercontent.com/72164/82662591-213e0d00-9bfc-11ea-82af-9d06ede3798f.png">

**Comment by hugovk** [comment]
🍰 Happy 10th birthday to Pillow!

* 2010-07-31 https://mail.python.org/pipermail/image-sig/2010-July/006423.html

🎂 And a belated happy 25th to the original PIL!

* 1995-06-15 https://mail.python.org/pipermail/image-sig/2005-June/003377.html

**Comment by aclark4life** [comment]
Not sure when it happened, but @radarhere takes the commit-lead!! 

<img width="974" alt="Screen Shot 2020-07-31 at 11 01 06 AM" src="https://user-images.githubusercontent.com/72164/89048055-472f0e80-d31d-11ea-813a-26022d393f64.png">

**Comment by hugovk** [comment]
5,000 issues and PRs!

https://github.com/python-pillow/Pillow/pull/5000

More numbers:

![image](https://user-images.githubusercontent.com/1324225/96874840-3eaf2580-147f-11eb-8bcd-b3462b385890.png)

**Comment by radarhere** [comment]
Something weird has happened with the PyPI download numbers - we're now at 30 million downloads in the last month!

**Comment by hugovk** [comment]
> Something weird has happened with the PyPI download numbers - we're now at 30 million downloads in the last month!

I think the the old numbers were missing some downloads, and the new ones should be better/more reliable. e.g. https://github.com/crflynn/pypistats.org/issues/34#issuecomment-816184362

**Comment by hugovk** [comment]
Mars helicopters using Pillow: 1

* https://github.blog/2021-04-19-open-source-goes-to-mars/
* https://docs.github.com/en/github/setting-up-and-managing-your-github-profile/personalizing-your-profile#list-of-qualifying-repositories-for-mars-2020-helicopter-contributor-badge

![image](https://user-images.githubusercontent.com/1324225/115297043-63e9cb80-a164-11eb-8240-1b20a354c7ba.png)

**Comment by aclark4life** [comment]
OMG

**Comment by wiredfool** [comment]
Snakes in spaaaace!

**Comment by wiredfool** [comment]
We're the second most popular "other" library in the 2020 jetbrains survey: https://www.jetbrains.com/lp/python-developers-survey-2020/

That doesn't sound like much, except that we're 3/5 as popular as Requests (the first in the "other" category) and used by 32% of respondents.

**Comment by hugovk** [comment]
Comparing with earlier years:

* [2016](https://www.jetbrains.com/pycharm/python-developers-survey-2016/): 20%
* [2017](https://www.jetbrains.com/research/python-developers-survey-2017/): 15%
* [2018](https://www.jetbrains.com/research/python-developers-survey-2018/): 29%
* [2019](https://www.jetbrains.com/lp/python-developers-survey-2019/): 32%
* [2020](https://www.jetbrains.com/lp/python-developers-survey-2020/): 32%

**Comment by aclark4life** [comment]
[Sad news from Zurich](https://lwn.net/ml/python-dev/CAP7%2BvJLbxJY-SPPzaoT4qnhF9MRkXDuhmr8wxZ%3DEs1X2ob3gMg%40mail.gmail.com/) RIP Fredrik 🙏

**Comment by aclark4life** [comment]
[504 followers](https://twitter.com/PythonPillow/followers)!

<img width="1792" alt="Screen Shot 2022-01-24 at 1 50 35 PM" src="https://user-images.githubusercontent.com/72164/150845635-70ceca6b-9aa8-43fc-ab00-0343ed86c9a3.png">

**Comment by hugovk** [comment]
> Comparing with earlier years:
> 
> * [2016](https://www.jetbrains.com/pycharm/python-developers-survey-2016/): 20%
> * [2017](https://www.jetbrains.com/research/python-developers-survey-2017/): 15%
> * [2018](https://www.jetbrains.com/research/python-developers-survey-2018/): 29%
> * [2019](https://www.jetbrains.com/lp/python-developers-survey-2019/): 32%
> * [2020](https://www.jetbrains.com/lp/python-developers-survey-2020/): 32%

Pillow is second and 31% in the [Python Developers Survey 2021 Results](https://lp.jetbrains.com/python-developers-survey-2021/#FrameworksLibraries).

<img width="510" alt="image" src="https://user-images.githubusercontent.com/1324225/171954895-9f6fd13a-6789-4632-a2dd-957e924fc53e.png">

**Comment by radarhere** [comment]
10,000 stars

**Comment by hugovk** [comment]
<img width="783" alt="image" src="https://user-images.githubusercontent.com/1324225/182185116-64815bfe-128f-479e-8247-18b6970cb926.png">

https://star-history.com/#python-pillow/Pillow&Date

**Comment by radarhere** [comment]
Looks like #6281 caused us to jump 5%, for a new total of 96% code coverage.

**Comment by hugovk** [comment]
50 million monthly downloads! 

Specifically: 50,222,924 in January 2023, and number 70 in the list.

https://hugovk.github.io/top-pypi-packages/

**Comment by hugovk** [comment]
> Comparing with earlier years:
> 
> * [2016](https://www.jetbrains.com/pycharm/python-developers-survey-2016/): 20%
> * [2017](https://www.jetbrains.com/research/python-developers-survey-2017/): 15%
> * [2018](https://www.jetbrains.com/research/python-developers-survey-2018/): 29%
> * [2019](https://www.jetbrains.com/lp/python-developers-survey-2019/): 32%
> * [2020](https://www.jetbrains.com/lp/python-developers-survey-2020/): 32%
> * [2021](https://lp.jetbrains.com/python-developers-survey-2021/#FrameworksLibraries): 31%

Pillow is second and 29% in the [Python Developers Survey 2022 Results](https://lp.jetbrains.com/python-developers-survey-2022/#FrameworksLibraries).

<img width="560" alt="image" src="https://github.com/python-pillow/Pillow/assets/1324225/0578317d-5025-4e7e-a915-61a619427f99">

**Reply by aclark4life** [reply]
GO GO GO

**Reply by hugovk** [reply]
Pillow is second and 28% in the [Python Developers Survey 2023 Results](https://lp.jetbrains.com/python-developers-survey-2023/#frameworks-and-libraries).

![image](https://github.com/user-attachments/assets/ec897817-f4c3-4659-b968-9da58a3deb99)

**Comment by radarhere** [comment]
The Pillow Github page is so popular that the PyPI downloads badge can't handle the traffic.

![Screenshot 2023-10-30 at 4 48 10 pm](https://github.com/python-pillow/Pillow/assets/3112309/af81425c-fa36-4add-8828-c930ae26387a)

**Comment by radarhere** [comment]
400 contributors

**Comment by aclark4life** [comment]
Remind me to recursively celebrate the 9 year anniversary of #6636 later this year on October 27 😂

**Comment by aclark4life** [comment]
Many CVEs fixed over 30 years 🎉 !

- 1991-1998 (PIL): No [CVE List](https://www.cve.org/About/History)
- 1999-2009 (PIL): No known CVEs
- 2010-2018 (Pillow): [CVE-2014-1932](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2014-1932), [CVE-2014-3589](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2014-3589), [CVE-2016-0740](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0740), [CVE-2016-3076](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-3076)
- 2019-2024 (Pillow w/Tidelift support): [CVE-2019-16865](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-16865), [CVE-2019-19911](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-19911), [CVE-2020-10177](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-10177), [CVE-2020-15999](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15999), [CVE-2020-35653](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-35653), [CVE-2021-25289](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-25289), [CVE-2020-35654](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-35654), [CVE-2020-35654](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-35654), [CVE-2021-27921](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-27921), [CVE-2021-27922](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-27922), [CVE-2021-27923](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-27923), [CVE-2021-25287](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-25287), [CVE-2021-25288](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-25288), [CVE-2021-34552](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-34552), [CVE-2021-23437](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-23437), [CVE-2022-22817](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-22817), [CVE-2022-24303](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24303), [CVE-2022-30595](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-30595), [CVE-2023-44271](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-44271), [CVE-2023-4863](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-4863)

**Comment by aclark4life** [comment]
On January 2, 2020, [the PyPI moderators exhausted the PEP 541 process for contacting the PIL project owner](https://github.com/python-pillow/Pillow/issues/1535#issuecomment-570308446) and we accepted their offer to manage the PIL project on PyPI.

**Comment by aclark4life** [comment]
14 years after naming Pillow, considered acronyms for Pillow (although Pillow is already word so maybe not an acronym): https://github.com/python-pillow/Pillow/discussions/7992

**Comment by aclark4life** [comment]
First [archived repository](https://github.com/python-pillow/pillow-wheels)!

**Comment by hugovk** [comment]
October 2015:

> 10,511,869 downloads!

October 2024:

3,001,836,264 downloads!


https://www.pepy.tech/projects/pillow

**Comment by radarhere** [comment]
20,020 commits.

**Comment by radarhere** [comment]
21,000 commits
